R/supervised-model-deployment.R

```diff
@@ -89,16 +89,22 @@ SupervisedModelDeployment <- R6Class("SupervisedModelDeployment",
   self$params$debug <- p$debug
   
   if (!is.null(p$modelName))
-    self$params$modelName <- p$modelName
+  self$params$modelName <- p$modelName
 
   # for deploy method
   if (!is.null(p$cores))
   self$params$cores <- p$cores
+  
+  if (is.null(self$params$modelName))
+  self$params$modelName = private$modelName
   },
 
   loadData = function() {
-    cat('Loading Data...','\n')
+    # Load model info
+    cat('Loading Model Info...','\n')
+    private$loadModelAndInfo(private$modelName)
```

Comment:
The argument passed to loadModelAndInfo is used in the warning message to point people to the documentation and should be the full model name ("RandomForest", "LinearMixedModel", etc.) as opposed to the name used to save the model ("RF", "LMM", etc.). Maybe we should add another parameter -- modelType or something like that -- that can't be set by the user. This would also make it easy to still single out XGBoost if we decide to do binary classification or regression with XGBoost.